pkg/configmap/services/services.go

```diff
@@ -57,6 +68,119 @@ type FileInfo struct {
 	Checksum string `json:"checksum"`
 }
 
+// servicesCMData represents the Data field of a `windows-services` ConfigMap resource
+type servicesCMData struct {
+	// Services contains all data required to configure the required Windows services to configure an instance as a Node
+	Services []Service `json:"services"`
+	// Files the path and checksum of files copied to the instance by WMCO
+	Files []FileInfo `json:"files"`
+}
+
+// NewServicesConfigMapData returns a new servicesCMData. Validates given object contents on creation.
+func NewServicesConfigMapData(services *[]Service, files *[]FileInfo) (*servicesCMData, error) {
+	cmData := &servicesCMData{*services, *files}
+	if err := cmData.validate(); err != nil {
+		return nil, errors.Wrap(err, "unable to create services ConfigMap data object")
+	}
+	return cmData, nil
+}
+
+// Parse converts ConfigMap data into the objects representing a Windows services ConfigMap schema
+// Returns error if the given data is invalid in structure or contents.
+func Parse(data map[string]string) (*servicesCMData, error) {
+	if len(data) != 2 {
+		return nil, errors.New("services ConfigMap should have exactly 2 keys")
+	}
+
+	services := &[]Service{}
+	files := &[]FileInfo{}
+
+	value, ok := data[servicesKey]
+	if !ok {
+		return nil, errors.Errorf("expected `%s` key to exist", servicesKey)
+	}
+	if err := json.Unmarshal([]byte(value), services); err != nil {
+		return nil, errors.Wrapf(err, "`%s` must represent list of Service", servicesKey)
```

Comment:
a list of Services

Reply:
could just return error without wrapping it as i dont think you're adding much info here